src/vsr/replica.zig

```diff
@@ -1314,6 +1325,30 @@ pub fn ReplicaType(
                 .ping_timestamp_monotonic = message.header.ping_timestamp_monotonic,
                 .pong_timestamp_wall = @bitCast(self.clock.realtime()),
             }));
+
+            if (message.header.replica < self.replica_count) {
+                const upgrade_targets = &self.upgrade_targets[message.header.replica];
+                if (upgrade_targets.* == null or
+                    upgrade_targets.*.?.timestamp > message.header.ping_timestamp_monotonic)
```

Comment:
Huh, I think this is incorrect --- monotinic timestamps are not guaranteed to be monotinic across restarts. The doom scenario here would be a buggy replica advedtising a very high monotonic timestamp, then crashing, then restarting with a much smaller monotonic timestamp and not being able to sync versions. 

Not sure what's the best solution here, options I have in mind:

* use checkpoint_op/view for monotonicity (doesn't work for idle cluster)
* use op for monotonicity (better then checkpoint, but still doesnt' work for idle cluster)
* use real time --- that _should_ be stable across restarts, but is not monotinic overwise
* just always update the upgrade_target, making an assumption that all old messages in the network eventually die? 

I'd probably go for this combination:

* if the current target has higher view or checkpoint, skip the new target
* otherwise, just replace the target, ignoring the timestamp

I do not feel great about timestamps, far-future timestamp locking us up from an upgrade feels like a nasty failure mode.